Source/ASImageNode+AnimatedImage.mm

```diff
@@ -70,13 +71,20 @@ - (void)_locked_setAnimatedImage:(id <ASAnimatedImageProtocol>)animatedImage
       [self _locked_setShouldAnimate:YES];
     }
   } else {
-      // Clean up after ourselves.
-      self.contents = nil;
-      [self setCoverImage:nil];
+    // Clean up after ourselves.
+    shouldCleanup = YES;
   }
   
-  [self animatedImageSet:_animatedImage previousAnimatedImage:previousAnimatedImage];
+  {
+    // Unlock for cleanup and calling subclass hook
+    ASUnlockScope(self);
     
+    self.contents = nil;
+    [self setCoverImage:nil];
```

Comment:
Shouldn't we wrap the above lines in a `shouldCleanup` block?

Reply:
Hm, can you explain why `-setCoverImage:` needs to be called unlocked? 

Somewhat unrelated: that method acquires the lock right away, so shouldn't we just call `-_locked_setCoverImage:` directly here?